### PR TITLE
Modify ExAws.Dynamo.Encodable.encode for: List to handle encoding an …

### PR DESCRIPTION
**Pull request by rinosamakanata**
…all number List to Dynamo's NS(Number Set) type.

### CODE REVIEW

**Comment by rinosamakanata** [comment]
@benwilson512, Thank you for your advice! I'll implement a protocol for MapSet that has all numbers that are not duplicated.

**Comment by benwilson512** [comment]
Well that's the nice thing about a MapSet, by definition it won't have any duplicates. You obviously will have to ensure they're all numbers.

**Comment by rinosamakanata** [comment]
@benwilson512 Thanks! I reverted my previous commits and newly implemented a protocol for MapSet. How about this?

**Comment by rinosamakanata** [comment]
In addition, I added some code to convert string-only MapSet for Dynamo's SS (StringSet) type.

File: lib/ex_aws/dynamo/encodable.ex [hunk 1]
```diff
@@ -111,3 +111,29 @@ defimpl ExAws.Dynamo.Encodable, for: List do
...
+  def encode(mapset, _) do
+    cond do
+      MapSet.size(mapset) == 0 ->
+        raise "Cannot determine a proper data type for an empty MapSet"
+      Enum.all?(mapset, &is_number/1) ->
+        %{"NS" => number_set_to_list(mapset)}
+      Enum.all?(mapset, &is_binary/1) ->
+        %{"SS" => MapSet.to_list(mapset)}
+      true ->
+        raise "All elements in a MapSet must be only numbers or only strings"
```
Comment (benwilson512) [inline lib/ex_aws/dynamo/encodable.ex]: I'm not wildly enthusiastic about raising exceptions here, but i'm not sure that the alternatives are much better.

**Comment by rinosamakanata** [comment]
@benwilson512 Yes, I'm not sure about raising exceptions, but I didn't find better solution, too.. Thank you for merging!

**Pull request merged**